include/swift/Reflection/ReflectionContext.h

```diff
@@ -221,6 +221,22 @@ class ReflectionContext
             SectionsBuf + (I * sizeof(typename T::Section)));
         if (strncmp(S->sectname, Name.data(), strlen(Name.data())) != 0)
           continue;
+
+        // The above check verifies that `Name` is a prefix to the examined
+        // section name, to allow for matching of sections with a suffix
+        // like `_TEXT`, etc.
+        // "__swift5_proto" section name is a substring of "__swift5_protos",
+        // Ensure we don't return the latter when looking for the former.
```

Comment:
Should the above really be looking for prefixes? I would expect section names to be exact. Segment names sometimes need prefix matching (e.g. `__DATA` becomes `__DATA_CONST`) but I don't think section names do. (I realize you didn't write the prefix check, but the right move here might be to make it an exact check rather than doing this additional check.)

Reply:
The section name there is actually `__swift5_fieldmd`. It's just that MachO section names are 16 bytes, not arbitrary NUL-terminated strings. The `__TEXT` comes from what's immediately after. It's really common to mix these up, because almost all section names you encounter in the real world are less than 16 bytes and pad the end with zeroes.

I _think_ the right fix here is just to change `strlen(Name.data())` to `sizeof(S->sectname)`. That will compare up to the 16-byte limit.

Reply:
Ah, that makes sense, thank you for the explanation! 
Using `strlen(Name.data()) ` does indeed achieve the expected behavior.

Reply:
Excellent!